### PR TITLE
[ng] drop `platform_directives` and `platform_pipes` from example pubspecs

### LINKED ISSUE #624: [ng] examples: drop `platform_*` entries from pubspec
Update all example pubspec files, **removing**
```yaml
transformers:
    - angular2:
        platform_directives:
            - 'package:angular2/common.dart#COMMON_DIRECTIVES'
        platform_pipes:
            - 'package:angular2/common.dart#COMMON_PIPES'
```
because support for the `platform_*` entries is being deprecated.

See https://github.com/dart-lang/angular2/issues/363 for details.

cc @kwalrath

### PR DESCRIPTION
**Pull request by chalin**
Fixes #624, **removing**
```yaml
transformers:
    - angular2:
        platform_directives:
            - 'package:angular2/common.dart#COMMON_DIRECTIVES'
        platform_pipes:
            - 'package:angular2/common.dart#COMMON_PIPES'
```
because support for the `platform_*` entries is being deprecated.

See https://github.com/dart-lang/angular2/issues/363 for details.

This PR updates the pubspecs and the component code to explicitly list `CORE_DIRECTIVES`, or  `CORE_DIRECTIVES` or `CORE_PIPES` as needed.

### CODE REVIEW

File: examples/ng/doc/architecture/lib/hero_service.dart [hunk 1]
```diff
@@ -9,14 +9,14 @@ import 'logger_service.dart';
...
-  final List<Hero> heroes = [];
+  final heroes = <Hero>[];
 
   HeroService(this._logger, this._backendService);
 
   List<Hero> getHeroes() {
     _backendService.getAll(Hero).then((heroes) {
       _logger.log('Fetched ${heroes.length} heroes.');
-      this.heroes.addAll(heroes); // fill cache
+      this.heroes.addAll(heroes as List<Hero>); // fill cache
```
Comment (kwalrath) [inline examples/ng/doc/architecture/lib/hero_service.dart]: Why is `as List<Hero>` necessary? I'd rather type it explicitly than have to use `as`.

File: examples/ng/doc/forms/lib/hero_form_component.dart [hunk 2]
```diff
@@ -34,13 +35,13 @@ class HeroFormComponent {
...
   /// Returns a map of CSS class names representing the state of [control].
   Map<String, bool> controlStateClasses(NgControl control) => {
-    'ng-dirty': control.dirty ?? false,
-    'ng-pristine': control.pristine ?? false,
-    'ng-touched': control.touched ?? false,
-    'ng-untouched': control.untouched ?? false,
-    'ng-valid': control.valid ?? false,
-    'ng-invalid': control.valid == false
-  };
+        'ng-dirty': control.dirty ?? false,
```
Comment (kwalrath) [inline examples/ng/doc/forms/lib/hero_form_component.dart]: The indentation here looks too large. Is this what dartfmt output?

File: examples/ng/doc/lifecycle-hooks/lib/after_content_component.dart [hunk 3]
```diff
@@ -1,45 +1,49 @@
...
-// #enddocregion hooks
+  // #enddocregion hooks
   final LoggerService _logger;
 
   AfterContentComponent(this._logger) {
     _logIt('AfterContent constructor');
   }
 
-// #docregion hooks
+  // #docregion hooks
```
Comment (kwalrath) [inline examples/ng/doc/lifecycle-hooks/lib/after_content_component.dart]: Have you tested the output to make sure changing the indentation on these comments doesn't adversely affect the docs?

File: examples/ng/doc/dependency-injection/lib/heroes/hero_list_component_2.dart [hunk 4]
```diff
@@ -13,16 +13,17 @@ import 'hero_service.dart';
...
+    </div>''',
+  directives: const [CORE_DIRECTIVES],
+)
 class HeroListComponent {
   final List<Hero> heroes;
 
   // #docregion ctor
-  HeroListComponent(HeroService heroService)
-      : heroes = heroService.getHeroes();
+  HeroListComponent(HeroService heroService) : heroes = heroService.getHeroes();
```
Comment (kwalrath) [inline examples/ng/doc/dependency-injection/lib/heroes/hero_list_component_2.dart]: This seems long. What line length are you using? Have you confirmed that it displays OK at common browser widths?

File: examples/ng/doc/architecture/lib/hero_service.dart (hunk 1 above)
Comment (chalin) [inline examples/ng/doc/architecture/lib/hero_service.dart]: I think that the checks might have become stricter. The problem is that `_backendService .getAll()` returns a `Future<List>` ... i.e., `Future<List<dynamic>>`. Hence the necessity for the cast.

There is quite a few things that I don't like about this code, but I chose to do only minimal changes for now.

File: examples/ng/doc/forms/lib/hero_form_component.dart (hunk 2 above)
Comment (chalin) [inline examples/ng/doc/forms/lib/hero_form_component.dart]: Yes I ran dartfmt.

File: examples/ng/doc/lifecycle-hooks/lib/after_content_component.dart (hunk 3 above)
Comment (chalin) [inline examples/ng/doc/lifecycle-hooks/lib/after_content_component.dart]: It doesn't. I've done this countless times before :).

File: examples/ng/doc/dependency-injection/lib/heroes/hero_list_component_2.dart (hunk 4 above)
Comment (chalin) [inline examples/ng/doc/dependency-injection/lib/heroes/hero_list_component_2.dart]: No, I just ran the formatter from within WS.

File: examples/ng/doc/architecture/lib/hero_service.dart (hunk 1 above)
Comment (kwalrath) [inline examples/ng/doc/architecture/lib/hero_service.dart]: I thought maybe this was related to removing the explicit typing of `heroes` up above. But if not... carry on.

**Pull request merged**